File: atm.cpp
#include<conio.h>
#include<bits/stdc++.h>

using namespace std;

class atm{										
private:										
	long int account_No;
	string name;
	int PIN;
	double balance;
	string mobile_No;
public:										
	void setData(long int account_No_a, string name_a, int PIN_a, double balance_a, string mobile_No_a){
		account_No = account_No_a;
		name = name_a;
		PIN = PIN_a;
		balance = balance_a;
		mobile_No = mobile_No_a;
	}

	long int getAccountNo(){
		return account_No;
	}

	string getName(){
		return name;
	}

	int getPIN(){
		return PIN;
	}

	double getBalance(){
		return balance;
	}

	string getMobileNo(){
		return mobile_No;
	}

	void setMobile(string mob_prev, string mob_new){
		if (mob_prev == mobile_No){						
			mobile_No = mob_new;						
			cout << endl << "Successfully Updated Mobile no.";
		    getch();				//getch is to hold the screen ( until user press any key )																			
		}

		else{						
			cout << endl << "Incorrect !!! Old Mobile no";
			getch();			
		}
	}

	void cashWithDraw(int amount_a){
		if (amount_a > 0 && amount_a < balance){		
			balance -= amount_a;		
			cout << endl << "Please Collect Your Cash"<<endl;
			cout << endl << "Available Balance :" << balance;
			getch();				
		}
		else{
			cout << endl << "Invalid Input or Insufficient Balance";
			getch();			
		}
	}

	void moneyDebit(int amount_a){
		if (amount_a > 0 && amount_a < balance){		
			balance -= amount_a;		
			cout << endl << "Money Transfered successfully !"<<endl;
			cout << endl << "Your account has a debit by transfer of "<<amount_a<<endl;
			cout << endl << "Available Balance :" << balance;
			getch();				
		}
		else{
			cout << endl << "Invalid Input or Insufficient Balance";
			getch();			
		}
	}

	void moneyCredit(string to ,int amount_a){
			balance += amount_a;
			cout << endl << to<<"'s account has been credited with amount "<<amount_a<<endl;
			cout << endl << "Available Balance :" << balance;
			getch();				
	}
};

/////////////////////////////////////////////////////////////////////////////////////////////
int main(){
    vector<atm> v;
	int choice = 0, enterPIN,index,x,y;		
	string username,whom;
	char c,d;
    long int enterAccountNo;
// 	system("cls");

	atm user1;
	user1.setData(1010101, "Omprakash", 1111, 45000.90, "9087654321");
    v.push_back(user1);
    atm user2;
	user2.setData(2020202, "Uday", 2222, 42000.00, "6370203478");
    v.push_back(user2);
    atm user3;
	user3.setData(3030303, "Aditya", 3333, 52000.75, "9885673421");
    v.push_back(user3);
    atm user4;
	user4.setData(4040404, "Simran", 4444, 50000.50, "8867046737");
    v.push_back(user4);

	do{
// 		system("cls");
		cout << endl << "*Welcome to ATM**" <<endl;
		cout << endl << "Enter Your User name :";	
		cin >> username; // username to be entered as i.e user1 ,user2,user3,user4 
        c=username[4];
        x=c - 48 ;
        cout << endl << "Enter Your Account number :";	
		cin >> enterAccountNo;
		cout << endl << "Enter PIN :";			
		cin >> enterPIN;

		if ((enterAccountNo == v[x-1].getAccountNo()) && (enterPIN == v[x-1].getPIN())){
			do{
				int amount = 0;
				string oldMobileNo, newMobileNo;
				// system("cls");
				cout << endl << "** Welcome to ATM ***" << endl;
				cout << endl << "Select Options ";
				cout << endl << "1. Check Balance";
				cout << endl << "2. Cash withdraw";
				cout << endl << "3. Show User Details";
				cout << endl << "4. Update Mobile no.";
				cout << endl << "5. Transfer Money";
				cout << endl << "6. Exit" << endl;
				cin >> choice;						
				switch (choice){					
				case 1:							
					cout << endl << "Your current Bank balance is :" << v[x-1].getBalance();
					getch();
					break;
				case 2:							
					cout << endl << "Enter the amount : ";
					cin >> amount;
					v[x-1].cashWithDraw(amount);		
					break;
				case 3:									
					cout << endl << "* User Details are :- ";
					cout << endl << "-> Account no  : " << v[x-1].getAccountNo();
					cout << endl << "-> Name        : " << v[x-1].getName();
					cout << endl << "-> Balance     : " << v[x-1].getBalance();
					cout << endl << "-> Mobile No.  : " << v[x-1].getMobileNo();
					getch();
					break;
				case 4:								
					cout << endl << "Enter Old Mobile No. :";
					cin >> oldMobileNo;							
					cout << endl << "Enter New Mobile No. :";
					cin >> newMobileNo;							
					v[x-1].setMobile(oldMobileNo, newMobileNo);	
					break;
				case 5:
					cout<<endl<<"Enter username whom you want to transfer money :";
					cin>>whom;   // username to be entered as i.e user1 ,user2,user3,user4 
					d=whom[4];
        			y=d - 48 ;
					cout<<endl<<"Enter the amount : ";
					cin >> amount;
					v[x-1].moneyDebit(amount);
					v[y-1].moneyCredit(whom,amount);
					break;
				case 6:
					exit(0);						
				default:							
					cout << endl << "Enter Valid Data !!!";
				}

			} while (1);												
		}
		else{
			cout << endl << "User Details are Invalid !!! ";
			getch();
		}
	} while (1);															
	return 0;
}

File: atm.java
import java.util.ArrayList;
import java.util.Scanner;

class ATM {
    private long accountNo;
    private String name;
    private int pin;
    private double balance;
    private String mobileNo;

    public void setData(long accountNo, String name, int pin, double balance, String mobileNo) {
        this.accountNo = accountNo;
        this.name = name;
        this.pin = pin;
        this.balance = balance;
        this.mobileNo = mobileNo;
    }

    public long getAccountNo() {
        return accountNo;
    }

    public String getName() {
        return name;
    }

    public int getPin() {
        return pin;
    }

    public double getBalance() {
        return balance;
    }

    public String getMobileNo() {
        return mobileNo;
    }

    public void setMobile(String oldMobileNo, String newMobileNo) {
        if (oldMobileNo.equals(mobileNo)) {
            mobileNo = newMobileNo;
            System.out.println("\nSuccessfully Updated Mobile no.");
        } else {
            System.out.println("\nIncorrect !!! Old Mobile no");
        }
    }

    public void cashWithdraw(int amount) {
        if (amount > 0 && amount < balance) {
            balance -= amount;
            System.out.println("\nPlease Collect Your Cash");
            System.out.println("Available Balance: " + balance);
        } else {
            System.out.println("\nInvalid Input or Insufficient Balance");
        }
    }

    public void moneyDebit(int amount) {
        if (amount > 0 && amount < balance) {
            balance -= amount;
            System.out.println("\nMoney Transferred successfully !");
            System.out.println("Your account has a debit by transfer of " + amount);
            System.out.println("Available Balance: " + balance);
        } else {
            System.out.println("\nInvalid Input or Insufficient Balance");
        }
    }

    public void moneyCredit(String to, int amount) {
        balance += amount;
        System.out.println("\n" + to + "'s account has been credited with amount " + amount);
        System.out.println("Available Balance: " + balance);
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        ArrayList<ATM> accounts = new ArrayList<>();

        ATM user1 = new ATM();
        user1.setData(1010101, "Omprakash", 1111, 45000.90, "9087654321");
        accounts.add(user1);

        ATM user2 = new ATM();
        user2.setData(2020202, "Uday", 2222, 42000.00, "6370203478");
        accounts.add(user2);

        ATM user3 = new ATM();
        user3.setData(3030303, "Aditya", 3333, 52000.75, "9885673421");
        accounts.add(user3);

        ATM user4 = new ATM();
        user4.setData(4040404, "Simran", 4444, 50000.50, "8867046737");
        accounts.add(user4);

        do {
            System.out.println("\n*Welcome to ATM**");
            System.out.print("\nEnter Your User name: ");
            String username = scanner.next(); // username to be entered as i.e user1, user2, user3, user4
            char c = username.charAt(4);
            int x = Character.getNumericValue(c);
            System.out.print("\nEnter Your Account number: ");
            long enterAccountNo = scanner.nextLong();
            System.out.print("\nEnter PIN: ");
            int enterPIN = scanner.nextInt();

            if (enterAccountNo == accounts.get(x - 1).getAccountNo() && enterPIN == accounts.get(x - 1).getPin()) {
                do {
                    int choice, amount;
                    String oldMobileNo, newMobileNo, whom;
                    System.out.println("\n** Welcome to ATM ***");
                    System.out.println("Select Options ");
                    System.out.println("1. Check Balance");
                    System.out.println("2. Cash withdraw");
                    System.out.println("3. Show User Details");
                    System.out.println("4. Update Mobile no.");
                    System.out.println("5. Transfer Money");
                    System.out.println("6. Exit");

                    choice = scanner.nextInt();

                    switch (choice) {
                        case 1:
                            System.out.println("\nYour current Bank balance is: " + accounts.get(x - 1).getBalance());
                            break;
                        case 2:
                            System.out.print("\nEnter the amount: ");
                            amount = scanner.nextInt();
                            accounts.get(x - 1).cashWithdraw(amount);
                            break;
                        case 3:
                            System.out.println("\n* User Details are :- ");
                            System.out.println("-> Account no  : " + accounts.get(x - 1).getAccountNo());
                            System.out.println("-> Name        : " + accounts.get(x - 1).getName());
                            System.out.println("-> Balance     : " + accounts.get(x - 1).getBalance());
                            System.out.println("-> Mobile No.  : " + accounts.get(x - 1).getMobileNo());
                            break;
                        case 4:
                            System.out.print("\nEnter Old Mobile No.: ");
                            oldMobileNo = scanner.next();
                            System.out.print("\nEnter New Mobile No.: ");
                            newMobileNo = scanner.next();
                            accounts.get(x - 1).setMobile(oldMobileNo, newMobileNo);
                            break;
                        case 5:
                            System.out.print("\nEnter username whom you want to transfer money: ");
                            whom = scanner.next(); // username to be entered as i.e user1, user2, user3, user4
                            char d = whom.charAt(4);
                            int y = Character.getNumericValue(d);
                            System.out.print("\nEnter the amount: ");
                            amount = scanner.nextInt();
                            accounts.get(x - 1).moneyDebit(amount);
                            accounts.get(y - 1).moneyCredit(whom, amount);
                            break;
                        case 6:
                            System.exit(0);
                        default:
                            System.out.println("\nEnter Valid Data !!!");
                    }

                } while (true);
            } else {
                System.out.println("\nUser Details are Invalid !!! ");
            }
        } while (true);
    }
}
